components/script/dom/htmlslotelement.rs

```diff
@@ -323,6 +333,15 @@ impl HTMLSlotElement {
             slottable.set_assigned_slot(DomRoot::from_ref(self));
         }
     }
+
+    /// <https://dom.spec.whatwg.org/#signal-a-slot-change>
+    pub(crate) fn signal_a_slot_change(&self) {
+        // Step 1. Append slot to slot’s relevant agent’s signal slots.
+        ScriptThread::add_signal_slot(self);
+
+        // Step 2. Queue a mutation observer microtask.
+        MutationObserver::queue_mutation_observer_microtask();
+    }
```

Comment:
I think this belong as a method like `ScriptThread::signal_a_slot_change` and it would be a good opportunity to move `queue_mutation_observer_microtask` to the script thread as well.